src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/IntegralMutatorFactory.java

```diff
@@ -269,6 +289,35 @@ protected final long mutateImpl(long value, PseudoRandom prng) {
       return value;
     }
 
+    protected final long crossOverImpl(long x, long y, PseudoRandom prng) {
+      switch (prng.indexIn(4)) {
+        case 0:
+          return mean(x, y);
+        case 1:
+          return forceInRange(x | y);
+        case 2:
+          return forceInRange(x & y);
+        case 3:
+          return forceInRange(x ^ y);
+        case 4:
+          return bitmask(x, y, prng);
+        default:
+          throw new AssertionError("Invalid cross over function.");
+      }
+    }
+
+    private long bitmask(long x, long y, PseudoRandom prng) {
+      long mask = prng.nextLong();
+      return forceInRange((x & mask) | (y & ~mask));
+    }
+
+    private static long mean(long x, long y) {
+      // Prevent overflows of large values in the naive implementation.
+      long xor = x ^ y;
+      long roundedDown = (x & y) + (xor >> 1);
+      return roundedDown + (1 & xor & (roundedDown >>> 31));
```

Comment:
This looks pretty tricky. Could you add some explanatory link? Why can't we add the two values shifted down by one?

Reply:
Code is based on a discussion from https://stackoverflow.com/questions/16115501/mean-of-two-ints-or-longs-without-overflow-truncating-towards-0

Basically supporting positive and negative numbers and rounding towards zero, which seemed beneficial in our case. Added a comment. We could also remove the rounding, if that is too tricky.